patches/server/0969-New-specific-event-watching-command.patch

```diff
@@ -0,0 +1,196 @@
++        List<Wrapper> wrappers = watching.get(event.getClass().getName());
++        if (wrappers != null)
++            wrappers.add(new Wrapper(event, listener));
++    }
++
++    @Override
++    public boolean execute(CommandSender sender, String subCommand, String[] args) {
++        if (args.length < 1) {
++            sender.sendMessage(text("Invalid Usage! /paper watchevent [event name]", RED));
++            return true;
++        }
++        List<Wrapper> wrappers = watching.get(args[0]);
++        if (wrappers != null) {
++            try {
++                dumpToFile(sender, args[0], wrappers);
++            } catch (IOException ex) {
++                sender.sendMessage(text("Failed to log event information! See the console for more info.", RED));
++                MinecraftServer.LOGGER.warn("Error occurred while logging event information", ex);
++            }
++        } else {
++            watching.put(args[0], new ArrayList<>());
++            sender.sendMessage(text("Now watching event " + args[0] + "!", GREEN));
++        }
++        return true;
++    }
++
++    @Override
++    public List<String> tabComplete(CommandSender sender, String subCommand, String[] args) {
++        return switch (args.length) {
++            case 0 -> new ArrayList<>(DumpListenersCommand.eventClassNames());
++            case 1 -> DumpListenersCommand.eventClassNames().stream()
++                .filter(clazz -> clazz.toLowerCase(Locale.ROOT).contains(args[0].toLowerCase(Locale.ROOT)))
++                .toList();
++            default -> Collections.emptyList();
++        };
++    }
++
++    private void dumpToFile(CommandSender sender, String event, List<Wrapper> wrappers) throws IOException {
++        JsonArray array = new JsonArray();
++        wrappers.forEach(w -> array.add(w.serialize()));
++
++        StringWriter stringWriter = new StringWriter();
++        JsonWriter jsonWriter = new JsonWriter(stringWriter);
++        jsonWriter.setIndent(" ");
++        jsonWriter.setLenient(false);
++        Streams.write(array, jsonWriter);
++
++        String[] unformatted = event.split("\\.");
++        String eventName = unformatted[unformatted.length - 1];
++
++        Path parent = PARENT.resolve(eventName);
++        Path path = parent.resolve("event-" + eventName + "-" +
++            FORMATTER.format(LocalDateTime.now()) + ".txt");
++        Files.createDirectories(parent);
++        Path file = Files.createFile(path);
++
++        try (PrintStream out = new PrintStream(Files.newOutputStream(file), false, StandardCharsets.UTF_8)) {
++            out.print(stringWriter);
++        }
++        sender.sendMessage(text("Successfully dumped event information to " + file, GREEN));
++        watching.remove(event);
++    }
++
++    /**
++     * EventWrapper subclass to apply listener information to our Json object
++     */
++    private static final class Wrapper extends EventWrapper {
```

Comment:
Does this also need its own override for shouldSerialize()? I think both wrapper should have the same logic for that, so you can put it in the superclass.